code/modules/mob/living/simple_animal/hostile/megafauna/drake.dm

```diff
@@ -104,70 +105,105 @@ Difficulty: Medium
+		QDEL_IN(summoned_object, 66)
+	while(amount > 0)
+		if(!target)
+			return
+		var/list/turfs = RANGE_TURFS(2, center)
+		var/turf/empty = pick((RANGE_TURFS(2, target) - RANGE_TURFS(1, target)) & turfs) // picks a turf within 2 of you not outside or in the shield
+		for(var/turf/T in turfs)
+			if(T != empty)
+				new /obj/effect/temp_visual/target(T)
+		amount--
+		sleep(22)
+	
+/mob/living/simple_animal/hostile/megafauna/dragon/proc/fire_cone()
 	playsound(get_turf(src),'sound/magic/fireball.ogg', 200, 1)
-
-	for(var/d in GLOB.cardinals)
-		INVOKE_ASYNC(src, .proc/fire_wall, d)
-
-/mob/living/simple_animal/hostile/megafauna/dragon/proc/fire_wall(dir)
-	var/list/hit_things = list(src)
-	var/turf/E = get_edge_target_turf(src, dir)
-	var/range = 10
-	var/turf/previousturf = get_turf(src)
-	for(var/turf/J in getline(src,E))
-		if(!range || (J != previousturf && (!previousturf.atmos_adjacent_turfs || !previousturf.atmos_adjacent_turfs[J])))
+	var/range = 15
+	var/list/turfs = list()
+	turfs = line_target(-40, range)
+	INVOKE_ASYNC(src, .proc/fire_line, turfs)
+	turfs = line_target(0, range)
+	INVOKE_ASYNC(src, .proc/fire_line, turfs)
+	turfs = line_target(40, range)
+	INVOKE_ASYNC(src, .proc/fire_line, turfs)
+	
+/mob/living/simple_animal/hostile/megafauna/dragon/proc/line_target(var/offset, var/range)
+	var/angle = ATAN2(target.x - src.x, target.y - src.y) + offset
+	var/turf/T = get_turf(src)
+	for(var/i in 1 to range)
+		var/turf/check = locate(src.x + cos(angle) * i, src.y + sin(angle) * i, src.z)
+		if(!check)
+			return
+		T = check
+	return getline(src, T)
+	
+/mob/living/simple_animal/hostile/megafauna/dragon/proc/fire_line(var/list/turfs)
+	for(var/turf/T in turfs)
+		if(istype(T, /turf/closed))
 			break
-		range--
-		new /obj/effect/hotspot(J)
-		J.hotspot_expose(700,50,1)
-		for(var/mob/living/L in J.contents - hit_things)
+		new /obj/effect/hotspot(T)
+		T.hotspot_expose(700,50,1)
+		for(var/mob/living/L in T.contents)
 			if(istype(L, /mob/living/simple_animal/hostile/megafauna/dragon))
 				continue
-			L.adjustFireLoss(20)
+			L.adjustFireLoss(30)
 			to_chat(L, "<span class='userdanger'>You're hit by the drake's fire breath!</span>")
-			hit_things += L
-		previousturf = J
-		sleep(1)
+			
+		// deals damage to mechs
+		for(var/obj/mecha/M in T.contents)
+			M.take_damage(60, BRUTE, "melee", 1)
+		sleep(1.25)
 
 /mob/living/simple_animal/hostile/megafauna/dragon/proc/triple_swoop()
-	swoop_attack(swoop_duration = 30)
-	swoop_attack(swoop_duration = 30)
-	swoop_attack(swoop_duration = 30)
+	swoop_attack()
+	swoop_attack()
+	swoop_attack(TRUE, null, 20) // longer cooldown on triple swoop ending
 
-/mob/living/simple_animal/hostile/megafauna/dragon/proc/swoop_attack(fire_rain, atom/movable/manual_target, swoop_duration = 40)
+/mob/living/simple_animal/hostile/megafauna/dragon/proc/swoop_attack(lava_pool = TRUE, atom/movable/manual_target, var/cooldown = 20)
```

Comment:
the cooldown is in deciseconds, currently you've set it to 2 seconds

Reply:
I do want it at 2 seconds, 20 seconds would be a bit excessive.

Reply:
> it's for player drakes, and 2 seconds is absolutely bonkers.

It's not for player drakes, player drakes have their own seperate cooldown timer now.